### PR TITLE
get data details genres

### PR DESCRIPTION
**Pull request by Namvn97**
https://edu-redmine.framgia.vn/issues/5968
get data details genres

### CODE REVIEW

**Comment by Namvn97** [comment]
ready

File: app/src/main/java/framgia/phannam/android/nmusic/data/source/sqlite/TrackDatabaseHelper.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            sTrackDatabaseHelper = new TrackDatabaseHelper(context);
+        }
+        return sTrackDatabaseHelper;
+    }
+
+    public TrackDatabaseHelper(Context context) {
+        super(context, DATABASE_NAME, null, DATABASE_VERSION);
+    }
+
+
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/phannam/android/nmusic/data/source/sqlite/TrackDatabaseHelper.java]: remove bl

File: app/src/main/java/framgia/phannam/android/nmusic/data/source/sqlite/TrackDatabaseHelper.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    public void removeTrackFromFavorite(Track track) {
+        SQLiteDatabase database = getWritableDatabase();
+        String selection = String.format(Constants.DB_QUERY_EQUAL_SELECTION, TrackEntity.ID);
+        String[] selectionArgs = {String.valueOf(track.getId())};
+        database.delete(TrackEntry.TABLE_NAME_FAVORITE, selection, selectionArgs);
+        database.close();
+    }
+
+    private Track parseTrackFromRow(Cursor cursor) {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/phannam/android/nmusic/data/source/sqlite/TrackDatabaseHelper.java]:
```suggestion
    private Track parseTrackFromRaw(Cursor cursor) {
```

File: app/src/main/java/framgia/phannam/android/nmusic/data/source/sqlite/TrackDatabaseHelper.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            track.setPublisherAlbumTitle(cursor.getString(cursor.getColumnIndexOrThrow(TrackEntity.ARTIST)));
+
+            int downloadable = cursor.getInt(cursor.getColumnIndexOrThrow(TrackEntity.DOWNLOADABLE));
+            if (downloadable == 1) {
+                track.setDownloadable(true);
+            } else {
+                track.setDownloadable(false);
+            }
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/phannam/android/nmusic/data/source/sqlite/TrackDatabaseHelper.java]: not print. handle Ex

File: app/src/main/res/layout/item_details_genres.xml [hunk 4]
```diff
@@ -2,20 +2,41 @@
...
+        <!--app:layout_constraintStart_toStartOf="parent"-->
+        <!--app:layout_constraintTop_toTopOf="parent" />-->
+    <android.support.v7.widget.CardView
+        android:id="@+id/card_details"
+        android:layout_width="@dimen/dp_70"
+        android:layout_height="@dimen/dp_70"
+        android:elevation="@dimen/dp_4"
+        app:cardCornerRadius="@dimen/dp_4"
+        app:cardUseCompatPadding="true"
+        android:layout_marginBottom="8dp"
```
Comment (NguyenMinhTuanB) [inline app/src/main/res/layout/item_details_genres.xml]: hardcode

**Comment by Namvn97** [comment]
ready

**Review by NguyenMinhTuanB — APPROVED** [review]

**Pull request merged**